testsuite/MDAnalysisTests/analysis/test_pca.py

```diff
@@ -145,6 +151,113 @@ def test_transform_universe():
     pca_test.transform(u2)
 
 
+def test_project_no_pca_run(u, pca):
+    pca_class = PCA(u, select=SELECTION)
+    with pytest.raises(ValueError) as exc:
+        pca_class.project_single_frame()
+    assert 'Call run() on the PCA before projecting' in str(exc.value)
+
+
+def test_project_none_anchor(u, pca):
+    group = u.select_atoms('resnum 1')
+    with pytest.raises(ValueError) as exc:
+        func = pca.project_single_frame(0, group=group, anchor=None)
+    assert ("'anchor' cannot be 'None'" +
+            " if 'group' is not 'None'") in str(exc.value)
+
+
+def test_project_more_anchor(u, pca):
+    group = u.select_atoms('resnum 1')
+    with pytest.raises(ValueError) as exc:
+        project = pca.project_single_frame(0, group=group, anchor='backbone')
+    assert "More than one 'anchor' found in residues" in str(exc.value)
+
+
+def test_project_less_anchor(u, pca):
+    group = u.select_atoms('all')
+    with pytest.raises(ValueError) as exc:
+        project = pca.project_single_frame(0, group=group, anchor='name CB')
+    assert ("Some residues in 'group'" +
+            " do not have an 'anchor'") in str(exc.value)
+
+
+def test_project_invalid_anchor(u):
+    pca = PCA(u, select='name CA').run()
+    group = u.select_atoms('all')
+    with pytest.raises(ValueError) as exc:
+        project = pca.project_single_frame(0, group=group, anchor='name N')
+    assert "Some 'anchors' are not part of PCA class" in str(exc.value)
+
+
+@pytest.mark.xfail
+def test_project_compare_projections(u_project):
+    # projections along different PCs should be different
+    pca = PCA(u_project, select=SELECTION).run()
+    project0 = pca.project_single_frame(0)
+    project1 = pca.project_single_frame(1)
+
+    u_project.trajectory[0]
+    coord0 = project0(u_project.trajectory.ts).positions
+    u_project.trajectory[0]
+    coord1 = project1(u_project.trajectory.ts).positions
+    assert_almost_equal(coord0, coord1, decimal=5)
+
+
+def test_project_reconstruct_whole(u, u_project):
+    # structure projected along all PCs
+    # should be same as the original structure
+    pca = PCA(u_project, select=SELECTION).run()
+    project = pca.project_single_frame()
+
+    coord_original = u.trajectory.ts.positions
+    coord_reconstructed = project(u_project.trajectory.ts).positions
+    assert_almost_equal(coord_original, coord_reconstructed, decimal=5)
```

Comment:
recommended (from https://numpy.org/doc/stable/reference/routines.testing.html )
```suggestion
    assert_allclose(coord_original, coord_reconstructed, rtol=1e-5)
```
We still have mostly assert_almost_equal but eventually we are going to replace them.

Reply:
Done